`TestServer_generateAgentVersionReport` flakiness

## Failure

#### Link(s) to logs

- https://github.com/gravitational/teleport.e/actions/runs/15362587744/job/43231549164

#### Relevant snippet

```
=== RUN   TestServer_generateAgentVersionReport/reloaded_and_terminating_instances_are_ignored
{"timestamp":"2025-05-31T11:34:37Z","level":"debug","caller":"auth/agent_version_report.go:146","message":"Collecting agent versions from inventory"}
{"timestamp":"2025-05-31T11:34:37Z","level":"debug","caller":"auth/agent_version_report.go:154","message":"Building the agent version report"}
    agent_version_report_test.go:269: 
        	Error Trace:	/__w/teleport.e/teleport.e/lib/auth/agent_version_report_test.go:269
        	Error:      	Should be empty, but was   (*autoupdate.AutoUpdateAgentReportSpec)(Inverse(protocmp.Transform, protocmp.Message{
        	            	  	"@type": s"teleport.autoupdate.v1.AutoUpdateAgentReportSpec",
        	            	  	"groups": map[string]protocmp.Message{
        	            	  		"default": {
        	            	  			"@type": s"teleport.autoupdate.v1.AutoUpdateAgentReportSpecGroup",
        	            	  			"versions": map[string]protocmp.Message{
        	            	  				"1.2.3": {
        	            	  					"@type": s"teleport.autoupdate.v1.AutoUpdateAgentReportSpecGroupVersion",
        	            	- 					"count": int32(1),
        	            	+ 					"count": int32(2),
        	            	  				},
        	            	  			},
        	            	  		},
        	            	  	},
        	            	  	"timestamp": protocmp.Message{"@type": s"google.protobuf.Timestamp", "nanos": int32(13[628](https://github.com/gravitational/teleport.e/actions/runs/15362587744/job/43231549164#step:7:629)8290), "seconds": int64(1748691277)},
        	            	  }))
        	Test:       	TestServer_generateAgentVersionReport/reloaded_and_terminating_instances_are_ignored
--- FAIL: TestServer_generateAgentVersionReport/reloaded_and_terminating_instances_are_ignored (0.00s)
```

--- a/lib/auth/agent_version_report_test.go
+++ b/lib/auth/agent_version_report_test.go
@@ -257,6 +257,14 @@ func TestServer_generateAgentVersionReport(t *testing.T) {
 					UpdaterInfo:      &types.UpdaterV2Info{UpdaterStatus: status, UpdateGroup: fixture.updateGroup},
 				})
 				if fixture.goodbye != nil {
+					// Sending the message twice is a little hack to make sure that the auth received the goodbye at
+					// least once. Else, the auth might still be in the small frame when it received the message
+					// over the channel but has not stored it yet. When this happens the test becomes flaky.
+					// Ideally we would peek into the inventory internals and wait for the goodbye to be stored,
+					// but we don't have access to inventory's private field.
+					// Because the channel is not buffered and a single go routine is reading on the other side, we know
+					// that the routine is done processing the first messages if we can send the second.
+					stream.fakeMsg(fixture.goodbye)
 					stream.fakeMsg(fixture.goodbye)
 				}
 				t.Cleanup(stream.close)

